### DISCUSSION TITLE
brew installation failure on macOS Monterey 12.3.1

### DISCUSSION THREAD

**Discussion by gshults** [Homebrew/discussions#3182] [Getting started] [answered]
How do I recover from the following installation failure?

I'm on a 16" 2019 MacBook Pro. I did a copy/paste of the following command:

`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

The installation finished with the following error:

`Updating files: 100% (6359/6359), done.
HEAD is now at ff34638b32f gloox: update 1.0.24 bottle.
Error: Failed to link all completions, docs and manpages:
  Permission denied @ rb_file_s_symlink - (../../../Homebrew/completions/zsh/_brew, /usr/local/share/zsh/site-functions/_brew)
Failed during: /usr/local/bin/brew update --force --quiet
`

I've attached the complete install log as a .txt file.

-Gerrie Shults

[brew install log.txt](https://github.com/Homebrew/discussions/files/8485156/brew.install.log.txt)

**Comment by gromgit** [comment, marked as answer]
> ```
> Error: Failed to link all completions, docs and manpages: Permission denied @ rb_file_s_symlink - (../../../Homebrew/completions/zsh/_brew, /usr/local/share/zsh/site-functions/_brew)
> ```

What's the output of `ls -laR /usr/local/share/zsh`?

**Reply by gshults** [reply]
ls -laR /usr/local/share/zsh
total 0
drwxr-xr-x  3 root    wheel   96 Jun  7  2021 .
drwxrwxr-x  4 gerrie  admin  128 Jun  7  2021 ..
drwxr-xr-x  3 root    wheel   96 Jun  7  2021 site-functions

/usr/local/share/zsh/site-functions:
total 0
drwxr-xr-x  3 root  wheel  96 Jun  7  2021 .
drwxr-xr-x  3 root  wheel  96 Jun  7  2021 ..
lrwxr-xr-x  1 root  wheel  57 Jun  7  2021 _mullvad -> /Applications/Mullvad VPN.app/Contents/Resources/_mullvad

**Reply by gromgit** [reply]
There's your problem: the `zsh` directory and everything under it is owned by root. If the `Mullvad VPN` installer asked you for your password, that's the likely culprit that created all that as root.

`sudo chown -R gerrie: /usr/local/share/zsh`, then run the installer again just to be sure.

**Reply by gshults** [reply]
> There's your problem: the `zsh` directory and everything under it is owned by root. If the `Mullvad VPN` installer asked you for your password, that's the likely culprit that created all that as root.
> 
> `sudo chown -R gerrie: /usr/local/share/zsh`, then run the installer again just to be sure.

Success!
Thank you very much!

**Reply by simranjeetbal** [reply]
This helped me as well, just wanted to pass along my thanks!
